api/src/main/java/jakarta/data/metamodel/constraint/LikePatternRecord.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel.constraint;
+
+import jakarta.data.metamodel.restrict.Operator;
+
+import java.util.Objects;
+
+record LikePatternRecord(String string, boolean caseSensitive, Character escape)
+        implements Like {
+
+    public static final char CHAR_WILDCARD = '_';
+    public static final char STRING_WILDCARD = '%';
+    public static final char ESCAPE = '\\';
+
+    public LikePatternRecord {
+        Objects.requireNonNull(string, "Pattern must not be null");
+    }
+
+    public LikePatternRecord(String pattern) {
+        this(pattern, true, null);
+    }
+
+    public LikePatternRecord(String pattern, Character escape) {
+        this(pattern, true, escape);
+    }
+
+    public LikePatternRecord(String pattern, char charWildcard, char stringWildcard) {
+        this(translate(pattern, charWildcard, stringWildcard));
+    }
+
+    public LikePatternRecord(String pattern, char charWildcard, char stringWildcard, char escape) {
+        this(translate(pattern, charWildcard, stringWildcard), escape);
+    }
```

Comment:
The `translate` method will need to know which escape character is being used so that it can insert the correct one.
```suggestion
    public LikePatternRecord(String pattern, char charWildcard, char stringWildcard, char escape) {
        this(translate(pattern, charWildcard, stringWildcard, escape), escape);
    }
```